include/crocoddyl/core/integrator/euler.hpp

```diff
@@ -58,39 +56,51 @@ class IntegratedActionModelEulerTpl : public ActionModelAbstractTpl<_Scalar> {
   virtual void print(std::ostream& os) const;
 
  protected:
-  using Base::nr_;     //!< Dimension of the cost residual
-  using Base::nu_;     //!< Control dimension
-  using Base::state_;  //!< Model of the state
-  using Base::unone_;  //!< Zero control
-
- private:
-  boost::shared_ptr<DifferentialActionModelAbstract> differential_;
-  Scalar time_step_;
-  Scalar time_step2_;
-  bool with_cost_residual_;
-  bool enable_integration_;
+  using Base::control_;      //!< Control parametrization
+  using Base::controlData_;  //!< Data for the control parametrization
+  using Base::differential_;
+  using Base::enable_integration_;
+  using Base::has_control_limits_;  //!< Indicates whether any of the control limits are active
+  using Base::nr_;                  //!< Dimension of the cost residual
+  using Base::nu_;                  //!< Dimension of the control
+  using Base::state_;               //!< Model of the state
+  using Base::time_step2_;
+  using Base::time_step_;
+  using Base::u_lb_;   //!< Lower control limits
+  using Base::u_ub_;   //!< Upper control limits
+  using Base::unone_;  //!< Neutral state
+  using Base::with_cost_residual_;
+
+  void init();
 };
 
 template <typename _Scalar>
-struct IntegratedActionDataEulerTpl : public ActionDataAbstractTpl<_Scalar> {
+struct IntegratedActionDataEulerTpl : public IntegratedActionDataAbstractTpl<_Scalar> {
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
   typedef _Scalar Scalar;
   typedef MathBaseTpl<Scalar> MathBase;
-  typedef ActionDataAbstractTpl<Scalar> Base;
+  typedef IntegratedActionDataAbstractTpl<Scalar> Base;
   typedef typename MathBase::VectorXs VectorXs;
   typedef typename MathBase::MatrixXs MatrixXs;
 
   template <template <typename Scalar> class Model>
   explicit IntegratedActionDataEulerTpl(Model<Scalar>* const model) : Base(model) {
     differential = model->get_differential()->createData();
     const std::size_t ndx = model->get_state()->get_ndx();
+    const std::size_t nv = model->get_state()->get_nv();
     dx = VectorXs::Zero(ndx);
+    u_diff = VectorXs::Zero(model->get_nu_diff());
```

Comment:
`get_nu` would not give me the correct size. I need the size of u_diff, not the size of u_params. I cannot directly access `nu_` even if it is protected because here I am not in the model class, but in the data class.